system_tests/system_tests_sync/test_external_accounts.py

```diff
@@ -0,0 +1,150 @@
+    headers = {}
+    credentials.apply(headers)
+    response = request(
+        url="https://dns.googleapis.com/dns/v1/projects/{}".format(project_id),
+        headers=headers,
+    )
+
+    if response.status == 200:
+        return response.data
+
+
+def dns_access_client_library(request, project_id):
+    service = discovery.build("dns", "v1")
+    request = service.projects().get(project=project_id)
+    return request.execute()
+
+
+dns_access_funcs = [dns_access_direct]
+try:
+    from googleapiclient import discovery
+
+    dns_access_funcs.append(dns_access_client_library)
+except ImportError as e:
+    if sys.version_info[0] == 3:
+        raise e
+
+
+@pytest.fixture(params=dns_access_funcs)
+def dns_access(request, http_request):
+    def wrapper(project_id):
+        return request.param(http_request, project_id)
+
+    yield wrapper
+
+
+@pytest.fixture
+def oidc_credentials(service_account_file, http_request):
+    result = service_account.IDTokenCredentials.from_service_account_file(
+        service_account_file, target_audience=_AUDIENCE_OIDC
+    )
+    result.refresh(http_request)
+    yield result
+
+
+@pytest.fixture
+def service_account_info(service_account_file):
+    with open(service_account_file) as f:
+        yield json.load(f)
+
+
+# Our external accounts tests involve setting up some preconditions, setting a
+# credential file, and then making sure that our client libraries can work with
+# the set credentials.
+def get_project_dns(dns_access, project_id, credential_data):
+    with NamedTemporaryFile() as credfile:
+        credfile.write(json.dumps(credential_data).encode("utf-8"))
+        credfile.flush()
+        old_credentials = os.environ.get("GOOGLE_APPLICATION_CREDENTIALS")
+        os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = credfile.name
+        try:
+            # If our setup and credential file are correct,
+            # discovery.build should be able to establish these as the default credentials.
+            return dns_access(project_id)
+        finally:
+            if old_credentials:
+                os.environ["GOOGLE_APPLICATION_CREDENTIALS"] = old_credentials
+            else:
+                del os.environ["GOOGLE_APPLICATION_CREDENTIALS"]
+
+
+# This test makes sure that setting an accesible credential file
+# works to allow access to Google resources.
+def test_file_based_external_account(
+    oidc_credentials, service_account_info, dns_access
+):
+    with NamedTemporaryFile() as tmpfile:
+        tmpfile.write(oidc_credentials.token.encode("utf-8"))
+        tmpfile.flush()
+
+        assert get_project_dns(
+            dns_access,
+            service_account_info["project_id"],
+            {
+                "type": "external_account",
+                "audience": _AUDIENCE_OIDC,
+                "subject_token_type": "urn:ietf:params:oauth:token-type:jwt",
+                "token_url": "https://sts.googleapis.com/v1beta/token",
```

Comment:
Let's use the GA endpoint since we are going to GA officially:
`"token_url": "https://sts.googleapis.com/v1/token",`

Reply:
done